healthcare/v1/src/main/java/snippets/healthcare/datasets/DatasetCreate.java

```diff
@@ -0,0 +1,101 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package snippets.healthcare.datasets;
+
+// [START healthcare_create_dataset]
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.healthcare.v1.CloudHealthcare;
+import com.google.api.services.healthcare.v1.CloudHealthcare.Projects.Locations.Datasets;
+import com.google.api.services.healthcare.v1.CloudHealthcareScopes;
+import com.google.api.services.healthcare.v1.model.Dataset;
+import com.google.api.services.healthcare.v1.model.Operation;
+import java.io.IOException;
+import java.util.Collections;
+
+public class DatasetCreate {
+  private static final String DATASET_NAME = "projects/%s/locations/%s/datasets/%s";
+  private static final JsonFactory JSON_FACTORY = new JacksonFactory();
+  private static final NetHttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+
+  public static void datasetCreate(String projectId, String regionId, String datasetId)
+      throws IOException {
+    // String projectId = "your-project-id";
+    // String regionId = "us-central1";
+    // String datasetId = "your-dataset-id";
+
+    // Initialize the client, which will be used to interact with the service.
+    CloudHealthcare client = createClient();
+
+    // Configure the dataset to be created.
+    Dataset dataset = new Dataset();
+    dataset.setTimeZone("America/Chicago");
+
+    // Create request and configure any parameters.
+    String parentName = String.format("projects/%s/locations/%s", projectId, regionId);
+    Datasets.Create request = client.projects().locations().datasets().create(parentName, dataset);
+    request.setDatasetId(datasetId);
+
+    // Execute the request, wait for the operation to complete, and process the results.
+    try {
+      Operation operation = request.execute();
+      System.out.println(operation.toPrettyString());
+      while (operation.getDone() == null || !operation.getDone()) {
+        // Update the status of the operation with another request.
+        Thread.sleep(500); // Pause for 500ms between requests.
+        operation =
+            client
+                .projects()
+                .locations()
+                .datasets()
+                .operations()
+                .get(operation.getName())
+                .execute();
+      }
+      System.out.println("Dataset created. Response content: " + operation.getResponse());
+    } catch (Exception ex) {
+      System.out.printf("Error during request execution: %s\n", ex.toString());
+      ex.printStackTrace(System.out);
+    }
```

Comment:
Can you be more specific?
Is there any mitigation?
Do you need this at all, why not just let it propagate and mention that it might happen?

Reply:
This is sort of a generic CRUD request to the service so the exception could be a number of things. I don't think it would be possible to be more specific. The mitigation should be clear from the error message returned (e.g. if auth is the issue). I see some samples catch errors/exceptions and others don't. Do you have a preference either way?

Reply:
Java prefers that we don't catch and ignore errors.  If we are going to ignore it, then let it be caught by a parent.  That said, if it might happen, it's good to mention it.
https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions